build.gradle

```diff
@@ -753,12 +753,19 @@ project(':clients') {
   task determineCommitId {
     ext.commitId = "unknown"
     def takeFromHash = 16
-    if (file("$rootDir/.git/HEAD").exists()) {
+
+    // Allow build system to set the commit ID as .git/HEAD might not be correct
+    // if there are local commits added for build purposes.
+    def kafkaCommitId = System.getenv('KAFKA_COMMIT_ID')
```

Comment:
An alternative is `project.hasProperty("commitId")`. One would set it during gradle's invocation: `./gradlew -PcommitId=foo`. The main advantage is that it's consistent with other overrides from the command-line. Thoughts?

Also, we should document this in the `Readme.md`.